src/components/Map.tsx

```diff
@@ -0,0 +1,114 @@
+
+
+interface StateProps {
+    readonly location: LocationState;
+    readonly pizzaPlaces: PizzaPlaceState;
+}
+
+const App = ({location, pizzaPlaces}: StateProps) => {
+    const mapRef: any = useRef(null);
+    const markerRef: any = useRef(null);
+
+    // Get current location once on load
+    useEffect(() => {
+      store.dispatch(getCurrentLocation())
+      store.dispatch(getPizzaPlaces())
+  }, [])
+
+    // Initialize the map on first load
+    useEffect(() => {
+        function onMapClick (d: any) {
+            const latlng = {
+                latitude: d.latlng.lat,
+                longitude: d.latlng.lng
+            }
+            store.dispatch(setLocation(latlng))
+        }
+        mapRef.current = L.map('mapid');
+        L.tileLayer('http://stamen-tiles-{s}.a.ssl.fastly.net/toner-background/{z}/{x}/{y}.png', {
+            attribution: 'Map tiles by <a href="http://stamen.com">Stamen Design</a>, <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a> &mdash; Map data &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+            maxZoom: 18}).addTo(mapRef.current);
+        mapRef.current.on('click', onMapClick);
+    }, [])
+
+    // Add marker to map whenever location changes
+    useEffect(() => {
+        if (!("isPending" in location || "errorMessage" in location)) {
+            if (!markerRef.current) {
+                markerRef.current = L.circle([location.resource.latitude, location.resource.longitude], {
+                    color: 'blue',
+                    fillColor: '#f03',
+                    fillOpacity: 0.5,
+                    radius: 100
+                }).addTo(mapRef.current);
+            } else {
+                markerRef.current.setLatLng([location.resource.latitude, location.resource.longitude])
+            }
+            markerRef.current.bindPopup("Hey, that's me!")
+            mapRef.current.setView([location.resource.latitude, location.resource.longitude], 13);
+        }
+    }, [location])
+
+    // Add pizza places to map
+    useEffect(() => {
+      console.log("running")
+      function addPizzaToMap (p: PizzaPlace) {
+        L.marker([p.latitude, p.longitude])
+          .addTo(mapRef.current)
+          .bindPopup(p.name);
+      }
+      if (!("isPending" in pizzaPlaces || "errorMessage" in pizzaPlaces)) {
+        pizzaPlaces.resource.forEach(p => {
+          console.log(p)
+          addPizzaToMap(p)
+        })
+      }
+    }, [pizzaPlaces])
+
+
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        Pizza Map
+      </header>
+        <div className={classNames({
+          hidden: "isPending" in location ? true : false
+        })}>
+          <div id="mapid">
+          </div>
+        </div>
+
+        <div id="map-loading" className={classNames({
+          hidden: "isPending" in location ? false : true
+        })}>
+          Getting current location...
```

Comment:
It'd be good to account for errors here; if I reject the geolocation request I see a blank map but would expect to see an error message.